packages/paste-core/components/avatar/src/index.tsx

```diff
@@ -41,8 +42,43 @@ const AvatarContents: React.FC<AvatarContentProps> = ({name, size = DEFAULT_SIZE
   );
 };
 
+const colorVariants: Record<ColorVariants, BoxStyleProps> = {
+  default: {
+    backgroundColor: 'colorBackgroundUser',
+    color: 'colorText',
+  },
+  decorative10: {
+    backgroundColor: 'colorBackgroundDecorative10Weakest',
+    color: 'colorTextDecorative10',
+  },
+  decorative20: {
+    backgroundColor: 'colorBackgroundDecorative20Weakest',
+    color: 'colorTextDecorative20',
+  },
+  decorative30: {
+    backgroundColor: 'colorBackgroundDecorative30Weakest',
+    color: 'colorTextDecorative30',
+  },
+  decorative40: {
+    backgroundColor: 'colorBackgroundDecorative40Weakest',
+    color: 'colorTextDecorative40',
+  },
+};
+
+const shapeVariants: Record<ShapeVariants, BoxStyleProps> = {
```

Comment:
suggestion (non-blocking): I know we want to be transparent about which styles the prop will affect, but could the prop be called 'variant'? Not blocking but it feels weird to me to have `shape="user"`

Reply:
I don't disagree, I went with what was in the ticket - but I'll bring it up in standup tomorrow!

Reply:
mmmm, OK, yeah I can see that.

Essentially what happened is a few of us were discussing variants, and shoving too much into `variant` for Avatars. For a time is was just "add the square variant", then "add the decorative options like Badge". But square for entities and circles for users, could all be decorative colors and we didn't want 10 variants.

So we tried to abstract some overlapping stuff out of variant and _maaaaaaybe_ got carried away.

`user` and `entity` could probably be fine as variants, they could also be fine remaining as shapes.

Great call out though @nkrantz